app/src/main/java/com/islamey/hamza/wazaker/ui/CounterFragment/CounterFragment.kt

```diff
@@ -1,86 +1,81 @@
-    private var sharedPreferences: SharedPreferences? = null
+    private lateinit var binding: FragmentCounterBinding
 
     private var counts = 0
     private var totalCounts = 0
-
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
+        setHasOptionsMenu(true)
+        binding = FragmentCounterBinding.inflate(layoutInflater, container, false)
+        return binding.root
+    }
 
-        _binding = FragmentCounterBinding.inflate(layoutInflater, container, false)
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
 
-        getDataFromSharedPreferences()
+        initUi()
+    }
 
-        binding.counterscreen.setOnClickListener {
-            counts++
-            totalCounts++
+    private fun initUi() {
 
-            checkCounts(counts)
-            binding.countertextview.text = counts.toString()
-            setTotalZekerCountstext(totalCounts)
+        getDataFromSharedPreferences()
 
-            saveTotalcountsInSharedPreference()
+        binding.counterScreenLayout.setOnClickListener {
+            incrementCounter()
         }
 
         binding.resetbtn.setOnClickListener {
-            counts = 0
-            binding.countertextview.text = counts.toString()
-            binding.countertextview.setTextColor(Color.WHITE)
+            resetCounter()
         }
-
-
-        return binding.root
     }
 
     @SuppressLint("SetTextI18n")
     fun getDataFromSharedPreferences() {
-        sharedPreferences = context?.getSharedPreferences(MY_PREFS_NAME, Context.MODE_PRIVATE)
-        totalCounts = sharedPreferences!!.getInt("zekertotalcounts", totalCounts)
-        setTotalZekerCountstext(totalCounts)
+        totalCounts = context?.let { getTotalCounts(it) }!!
+        setTotalZekerCountsText(totalCounts)
     }
 
     @SuppressLint("SetTextI18n")
-    fun setTotalZekerCountstext(Totalzekercounts: Int) {
-        binding.totalCountstextview.text = getString(R.string.totalzeker) + "  " + Totalzekercounts
+    fun setTotalZekerCountsText(TotalZekerCounts: Int) {
+        binding.totalCountstextview.text = getString(R.string.totalzeker) + "  " + TotalZekerCounts
     }
 
-    private fun checkCounts(counts: Int) {
-        if (counts % 33 == 0) getRandomcolor()
-    }
+    private fun incrementCounter() {
+        counts++
+        totalCounts++
 
-    private fun getRandomcolor() {
-        val currentColor: Int
-        if (counts % 33 == 0) {
-            currentColor = RandomColor.getRandomColor()
-            binding.countertextview.setTextColor(currentColor)
-        }
+        checkCounts(counts)
+        binding.countertextview.text = counts.toString()
+        setTotalZekerCountsText(totalCounts)
+
+        saveTotalCountsInSharedPreference(requireContext(), totalCounts)
     }
 
+    private fun resetCounter() {
+        counts = 0
```

Comment:
this should not be done here, as when rotation occurs we will lose it.
please move to view model or bundle